packages/app/components/Settings/CourseInformation.tsx

```diff
@@ -0,0 +1,152 @@
+    setCrns(course.crns.map((c) => c.toString().padStart(5, "0")));
+  };
+
+  const handleSaveChanges = async () => {
+    const value = await form.validateFields();
+    value.crns = Array.from(new Set(crns));
+
+    try {
+      await API.course.editCourseInfo(course.id, value);
+      message.success("Successfully updated course information.");
+    } catch (e) {
+      message.error(e.response?.data?.message);
+    }
+  };
+
+  const handleCRNAdd = () => {
+    if (inputCRN) {
+      setCrns([...crns, inputCRN]);
+    }
+    setShowCRNInput(false);
+    setInputCRN(null);
+  };
+
+  const handleCRNDelete = (crn) => {
+    crns.splice(crns.indexOf(crn), 1);
+    setCrns(crns);
+  };
+
+  return (
+    <div>
+      <Space style={{ marginTop: "30px", marginBottom: "20px" }}>
+        <Button onClick={handleDiscardChanges}>Discard Changes</Button>
+
+        <Button onClick={handleSaveChanges} type="primary">
+          Save Changes
+        </Button>
+      </Space>
+
+      <Form form={form} layout="vertical" initialValues={course}>
+        <Form.Item
+          name="name"
+          label="Course Display Name"
+          tooltip="This is the course name that will be displayed within the app"
+          rules={[{ required: true, message: "Please input a display name." }]}
+        >
+          <Input placeholder="ex: CS 2500" maxLength={20} />
+        </Form.Item>
+
+        <Form.Item
+          name="coordinator_email"
+          label="Coordinator Email"
+          rules={[
+            {
+              required: true,
+              type: "email",
+              message: "Please input your email.",
+            },
+          ]}
+        >
+          <Input placeholder="example@northeastern.edu" />
+        </Form.Item>
+
+        <Form.Item
+          label="Office Hours Calendar URL"
+          tooltip={
+            <div>
+              See{" "}
+              <a
+                target="_blank"
+                rel="noreferrer"
+                href="https://info.khouryofficehours.com/coordinators-manual"
+              >
+                here
+              </a>{" "}
+              to create your office hours calendar
+            </div>
+          }
+          name="icalURL"
+          rules={[
+            {
+              required: true,
+              type: "url",
+              message: "Please input your office hours calendar URL.",
+            },
+          ]}
+        >
+          <Input placeholder="https://calendar.google.com/calendar/ical/.../basic.ics" />
+        </Form.Item>
+
+        <Form.Item label="Registered CRNs" name="crns">
+          {crns.map((crn) => (
+            <Tag
+              closeIcon={<DeleteTwoTone twoToneColor="#F76C6C" />}
+              key={crn}
+              closable={true}
+              onClose={() => handleCRNDelete(crn)}
+            >
+              {crn}
+            </Tag>
+          ))}
+          {showCRNInput ? (
+            <InputNumber<string>
```

Comment:
an input number is a bit weird because generally that is meant for incrementing a count little by little, i think you could just get away with a normal input but I wonder what others think

Reply:
I thought about this. I decided to go with number because it prevents the user from inputting any letters or symbols : |

Reply:
I'm pretty sure you can pass in a type as a prop to specify that it will be a number https://ant.design/components/input/